libminifi/src/core/ProcessGroup.cpp

```diff
@@ -200,7 +212,13 @@ void ProcessGroup::startProcessing(TimerDrivenSchedulingAgent& timeScheduler, Ev
 
 void ProcessGroup::stopProcessing(TimerDrivenSchedulingAgent& timeScheduler, EventDrivenSchedulingAgent& eventScheduler,
                                   CronDrivenSchedulingAgent& cronScheduler, const std::function<bool(const Processor*)>& filter) {
-  std::lock_guard<std::recursive_mutex> lock(mutex_);
```

Comment:
This and similar changes in the `startProcessingProcessors` and `startProcessing` functions are to avoid deadlock. It appeared in `ControllerServiceIntegrationTests` when `stop()` is called while the `getControllerService` is called in the processor's `onTrigger`. The `stop()`  function waits for the processor's `onTrigger` to finish, but it cannot finish the ProcessGroup's `findProcessor` method because it is waiting for the same mutex which is held here.

We may need to find a better solution as this may cause some concurrency issues I'm not currently aware of. I'm open for suggestions.